core/src/main/java/io/aiven/klaw/service/OperationalRequestsService.java

```diff
@@ -0,0 +1,337 @@
+    this.manageDatabase = manageDatabase;
+    this.mailService = mailService;
+    this.clusterApiService = clusterApiService;
+    this.commonUtilsService = commonUtilsService;
+    this.rolesPermissionsControllerService = rolesPermissionsControllerService;
+  }
+
+  public ApiResponse createConsumerOffsetsResetRequest(
+      ConsumerOffsetResetRequestModel consumerOffsetResetRequestModel)
+      throws KlawNotAuthorizedException {
+    log.info("createConsumerOffsetsResetRequest {}", consumerOffsetResetRequestModel);
+    checkIsAuthorized(PermissionType.REQUEST_CREATE_SUBSCRIPTIONS);
+    String userName = getUserName();
+    consumerOffsetResetRequestModel.setRequestor(userName);
+    consumerOffsetResetRequestModel.setRequestingTeamId(commonUtilsService.getTeamId(userName));
+
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+
+    OperationalRequest operationalRequest = new OperationalRequest();
+    copyProperties(consumerOffsetResetRequestModel, operationalRequest);
+    operationalRequest.setTenantId(commonUtilsService.getTenantId(userName));
+    if (operationalRequest.getOffsetResetType() == OffsetResetType.TO_DATE_TIME) {
+      operationalRequest.setResetTimeStamp(
+          getUTCTimeStamp(consumerOffsetResetRequestModel.getResetTimeStampStr()));
+    }
+    String result = dbHandle.requestForConsumerOffsetsReset(operationalRequest).get("result");
+
+    String requestFormattedStr = "Consumer group : " + operationalRequest.getConsumerGroup();
+    mailService.sendMail(
+        operationalRequest.getTopicname(),
+        requestFormattedStr,
+        "",
+        operationalRequest.getRequestor(),
+        operationalRequest.getApprover(),
+        operationalRequest.getRequestingTeamId(),
+        dbHandle,
+        MailType.RESET_CONSUMER_OFFSET_REQUESTED,
+        commonUtilsService.getLoginUrl());
+
+    return ApiResultStatus.SUCCESS.value.equals(result)
+        ? ApiResponse.ok(result)
+        : ApiResponse.notOk(result);
+  }
+
+  private Timestamp getUTCTimeStamp(String timeStampStr) {
+    DateTimeFormatter df = DateTimeFormatter.ofPattern(OFFSET_RESET_TIMESTAMP_FORMAT);
+    DateTimeFormatter dfUTC = df.withZone(ZoneOffset.UTC);
+
+    ZonedDateTime parsedDate = ZonedDateTime.parse(timeStampStr, dfUTC);
+    return Timestamp.from(parsedDate.toInstant());
+  }
+
+  public List<OperationalRequestsResponseModel> getConsumerOffsetsResetRequests(
+      String pageNo,
+      String currentPage,
+      OperationalRequestType operationalRequestType,
+      String requestStatus,
+      String env,
+      String wildcardSearch,
+      Order order,
+      boolean isMyRequest) {
+    log.debug("getTopicRequests page {} operationalRequestType {}", pageNo, operationalRequestType);
+    String userName = getUserName();
+    HandleDbRequests dbHandle = manageDatabase.getHandleDbRequests();
+    List<OperationalRequest> operationalRequests =
```

Comment:
Should we also have a check here for  checkIsAuthorized ?

Reply:
for getting requests, we dont have any checks. verified with other types too